service/src/main/java/bio/terra/profile/app/configuration/AzureConfiguration.java

```diff
@@ -63,11 +65,17 @@ public void setAuthorizedUserKey(String authorizedUserKey) {
   }
 
   public TokenCredential buildManagedAppCredentials() {
-    return new ClientSecretCredentialBuilder()
-        .clientId(managedAppClientId)
-        .clientSecret(managedAppClientSecret)
-        .tenantId(managedAppTenantId)
-        .build();
+    if (controlPlaneEnabled) {
```

Comment:
I think we might need test for this public method to ensure that different CredentialBuilders used.

Reply:
@sergiygetlin do you have an example of an existing test that I should look at? Seems like we use mocks for anything that would require a credential

Reply:
@aaronegrant I would try to split configuration and TokenCredential generation. Making TokenCredentialFactory and passing required boolean into the factory method.

Reply:
@sergiygetlin refactored to used ChainedTokenCredentialBuilder. Will first try to use a ManagedIdentity, and if that is not succesful, will fall through to use the historical ServicePrincipal auth. Creating a test for the MI auth path isn't doable since that auth method only works when running BPM in an Azure service. See the note in blue at https://learn.microsoft.com/en-us/azure/azure-app-configuration/howto-integrate-azure-managed-service-identity?tabs=core5x&pivots=framework-dotnet

Reply:
I agree that I don't see an adequate way to test this in a unit test. In the first iteration of this code, when there was a conditional, I could see a test that ensures the correct code path executes, but I think that is irrelevant when using the chained credential approach.